const factorial = require('../src/lib/factorial')

test('Если передано число с плавающией точкой, то должна выдаваться ошибка.', () => {
  expect(() => factorial(3.14)).toThrowError(
    new Error(
      'В качестве аргумента number принимаются только целые положительные числа.'
    )
  )
})

test('Если передана строка, то должна выдаваться ошибка.', () => {
  expect(() => factorial('foo')).toThrowError(
    new Error(
      'В качестве аргумента number принимаются только целые положительные числа.'
    )
  )
})

test('Если передано отрицательное число, то должна выдаваться ошибка.', () => {
  expect(() => factorial(-1)).toThrowError(
    new Error(
      'В качестве аргумента number принимаются только целые положительные числа.'
    )
  )
})

test('Если передан ноль, то должна выдаваться единица.', () => {
  expect(factorial(0)).toBe(1)
})

test('Если передано 2, то результат должен быть 2.', () => {
  expect(factorial(2)).toBe(2)
})

test('Если передано 4, то результат должен быть 24.', () => {
  expect(factorial(4)).toBe(24)
})

test('Если передано 8, то результат должен быть 40320.', () => {
  expect(factorial(8)).toBe(40320)
})
